### PR TITLE
Shared Stops Support

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing
### Description

This PR adds support for editing a project's shared stops config.

### CODE REVIEW

File: lib/manager/components/ProjectSettingsForm.js [hunk 1]
```diff
@@ -324,6 +326,25 @@ export default class ProjectSettingsForm extends Component<Props, State> {
             </ListGroupItem>
           </ListGroup>
         </Panel>
+        <Panel>
+          <Panel.Heading><Panel.Title componentClass='h4'>Shared Stops</Panel.Title></Panel.Heading>
+          <ListGroup>
+            <ListGroupItem>
+              <FormGroup>
+                {/* TODO: on enter, textarea should NOT submit */}
```
Comment (philip-cline) [inline lib/manager/components/ProjectSettingsForm.js]: I might add some weight to this `TODO` it feels a little strange hitting enter twice, but most interaction will be through copying and pasting or uploading a file so I don't think this is too serious.

File: lib/manager/components/ProjectSettingsForm.js [hunk 2]
```diff
@@ -324,6 +326,25 @@ export default class ProjectSettingsForm extends Component<Props, State> {
...
+                {/* TODO: Validate CSV? */}
```
Comment (philip-cline) [inline lib/manager/components/ProjectSettingsForm.js]: We should validate the CSV at least with the basic validation we do for transformations. Maybe some shared classes here?

File: lib/manager/components/ProjectSettingsForm.js [hunk 3]
```diff
@@ -324,6 +326,25 @@ export default class ProjectSettingsForm extends Component<Props, State> {
             </ListGroupItem>
           </ListGroup>
         </Panel>
+        <Panel>
+          <Panel.Heading><Panel.Title componentClass='h4'>Shared Stops</Panel.Title></Panel.Heading>
```
Comment (philip-cline) [inline lib/manager/components/ProjectSettingsForm.js]: New text should be i18n'd please :)

File: lib/manager/components/ProjectSettingsForm.js [hunk 4]
```diff
@@ -324,6 +326,25 @@ export default class ProjectSettingsForm extends Component<Props, State> {
...
+              <FormGroup>
+                {/* TODO: on enter, textarea should NOT submit */}
+                {/* TODO: Validate CSV? */}
+                <FormControl
+                  name={'sharedStopsConfig'}
+                  onChange={this._onChangeTextInput}
+                  value={model.sharedStopsConfig || ''}
+                  style={{height: model.sharedStopsConfig ? 200 : 150}}
+                  componentClass='textarea'
+                  placeholder='Shared stops config CSV'
```
Comment (philip-cline) [inline lib/manager/components/ProjectSettingsForm.js]: i18n please :)

**Review by philip-cline — CHANGES_REQUESTED** [review]
Can we just do some i18n and then this looks good assuming that we are limiting this PR in scope to just editing shared stops. I am looking forward to seeing how we handle displaying the validation on the front end

**Comment by miles-grant-ibigroup** [comment]
Thanks @philip-cline , changes are addressed

**Review by philip-cline — APPROVED** [review]
LGTM. Tests just failing bc the language files are mismatched now?

**Pull request merged**